.github/workflows/ci.yml

```diff
@@ -38,117 +37,82 @@ jobs:
+        uses: Swatinem/rust-cache@v2
       - uses: taiki-e/install-action@v2
         with: { tool: 'just' }
-      - uses: Swatinem/rust-cache@v2
-      - run: just env-info
-      - if: startsWith(matrix.runs-on, 'ubuntu')
-        name: Install linux dependencies
-        run: |
-          sudo apt-get update
-          # Make sure this list matches README
-          sudo apt-get install -y \
-            build-essential \
-            libcurl4-openssl-dev \
-            libglfw3-dev \
-            libjpeg-dev \
-            libpng-dev \
-            libsqlite3-dev \
-            libuv1-dev \
-            libwebp-dev \
-            libz-dev
-          if [ "${{ matrix.backend }}" = "opengl" ]; then
-            sudo apt-get install -y libopengl0
-          elif [ "${{ matrix.backend }}" = "vulkan" ]; then
-            sudo apt-get install -y mesa-vulkan-drivers glslang-dev
-          fi
-      - if: startsWith(matrix.runs-on, 'macos')
-        name: Install macOS dependencies
-        run: |
-          # Install dependencies via Homebrew
-          brew install \
-            curl \
-            glfw \
-            jpeg \
-            libpng \
-            sqlite \
-            libuv \
-            webp \
-            zlib
-          # For Vulkan on macOS, install MoltenVK
-          if [ "${{ matrix.backend }}" = "vulkan" ]; then
-            brew install molten-vk vulkan-headers
-          fi
-      # Setup rust version
-      - if: matrix.rust-version == 'stable'
-        run: rustup update stable && rustup default stable
-      - if: matrix.rust-version == 'msrv'
-        name: Read MSRV
+      - run: just env-info install-dependencies '${{ matrix.backend }}'
+      - name: Read MSRV
         id: msrv
-        run: echo "value=$(sed -nE 's/^rust-version[[:space:]]*=[[:space:]]*"([^"]+)".*/\1/p' Cargo.toml)" >> $GITHUB_OUTPUT
-      - if: matrix.rust-version == 'msrv'
-        name: Install Rust
+        run: echo "value=$(just get-msrv)" >> $GITHUB_OUTPUT
+      - name: Install MSRV Rust ${{ steps.msrv.outputs.value }}
         uses: dtolnay/rust-toolchain@stable
         with:
           toolchain: ${{ steps.msrv.outputs.value }}
-      # Run tests
-      - run: just ci-test ${{ matrix.backend }}
+      - run: just ci_mode=0 ci-test-msrv ${{ matrix.backend }}  # Ignore warnings in MSRV
 
   # This final step is needed to mark the whole workflow as successful
   # Don't change its name - it is used by the merge protection rules
   ci-passed:
+    needs: [ lint, test, test-msrv ]
     if: always()
-    needs: [ test, lint ]
-    name: CI Finished
     runs-on: ubuntu-latest
     steps:
       - name: Result of the needed steps
         run: echo "${{ toJSON(needs) }}"
-      - if: ${{ contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled') || contains(needs.*.result, 'skipped') }}
-        name: CI Result
+      - if: ${{ contains(needs.*.result, 'failure') || contains(needs.*.result, 'cancelled') }}
```

Comment:
I know this is currently correct, but If a skippable job (a job which depends on a step beforehand) is added, this breaks this job in the same way as martin was broken.

(🤷🏻‍♂️ .. grumble)